app/app_ecdsa.c

```diff
@@ -296,18 +314,34 @@ int app_ecdsa_handler(ACVP_TEST_CASE *test_case) {
         }
 
         sig_len = (size_t)i2d_ECDSA_SIG(sig_obj, &sig);
-        sig_ctx = EVP_MD_CTX_new();
-        if (!sig_ctx) {
-            printf("Error initializing sign CTX for ECDSA sigver\n");
-            goto err;
-        }            
 
-        if (EVP_DigestVerifyInit_ex(sig_ctx, NULL, md, NULL, NULL, pkey, NULL) != 1) {
-            printf("Error initializing signing for ECDSA sigver\n");
-            goto err;
-        }
-        if (EVP_DigestVerify(sig_ctx, sig, sig_len, tc->message, tc->msg_len) == 1) {
-            tc->ver_disposition = 1;
+        if (!tc->is_component) {
+            sig_ctx = EVP_MD_CTX_new();
+            if (!sig_ctx) {
+                printf("Error initializing sign CTX for ECDSA sigver\n");
+                goto err;
+            }
+
+            if (EVP_DigestVerifyInit_ex(sig_ctx, NULL, md, NULL, NULL, pkey, NULL) != 1) {
+                printf("Error initializing signing for ECDSA sigver\n");
+                goto err;
+            }
+            if (EVP_DigestVerify(sig_ctx, sig, sig_len, tc->message, tc->msg_len) == 1) {
+                tc->ver_disposition = 1;
+            }
+        } else {
+            comp_ctx = EVP_PKEY_CTX_new_from_pkey(NULL, pkey, NULL);
```

Comment:
Did comp_ctx just leaked from the previous assignment?

Reply:
the previous assignment is in a difference switch case for siggen instead of sigver and breaks; so we dont hit both branches in the same test case